src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -67,106 +65,93 @@ public StudyService(
+            Mono<String> caseFormat = getCaseFormat(uuid);
+            return Mono.zip(networkInfos, caseFormat)
+                    .flatMap(t -> {
+                        Study study = new Study(studyName, t.getT1().getNetworkUuid(), t.getT1().getNetworkId(), description, t.getT2(), uuid, true);
+                        return studyRepository.insert(study);
+                    });
+        });
     }
 
-    Study getStudy(String studyName) {
-        return studyRepository.findByName(studyName).orElse(null);
+    Mono<Study> getStudy(String studyName) {
+        return studyRepository.findByName(studyName);
     }
 
-    void deleteStudy(String studyName) {
-        Optional<Study> studyOpt = studyRepository.findByName(studyName);
-        if (studyOpt.isEmpty()) {
-            return;
-        }
-        Study study = studyOpt.get();
-        if (study.isCasePrivate()) {
-            try {
-                caseServerRest.delete("/" + CASE_API_VERSION + "/cases/{caseUuid}", study.getCaseUuid());
-            } catch (HttpStatusCodeException e) {
-                throw new StudyException("deleteStudy HttpStatusCodeException", e);
-            }
-        }
-        studyRepository.deleteByName(studyName);
-    }
+    Mono<Void> deleteStudy(String studyName) {
+        Mono<Study> studyMono = studyRepository.findByName(studyName);
 
-    List<CaseInfos> getCaseList() {
-        ParameterizedTypeReference<List<CaseInfos>> parameterizedTypeReference = new ParameterizedTypeReference<List<CaseInfos>>() { };
-        ResponseEntity<List<CaseInfos>> responseEntity;
-        try {
-            responseEntity = caseServerRest.exchange("/" + CASE_API_VERSION + "/cases",
-                    HttpMethod.GET,
-                    null,
-                    parameterizedTypeReference);
-
-            if (responseEntity.getStatusCode() == HttpStatus.OK) {
-                return responseEntity.getBody();
-            } else {
-                return Collections.emptyList();
+        return studyMono.flatMap(study -> {
+            String path = UriComponentsBuilder.fromPath(DELIMITER + CASE_API_VERSION + "/cases/{caseUuid}")
+                    .buildAndExpand(study.getCaseUuid())
+                    .toUriString();
+
+            if (study.isCasePrivate()) {
+                return webClient.delete()
+                        .uri(caseServerBaseUri + path)
+                        .retrieve()
+                        .bodyToMono(Void.class);
             }
-        } catch (HttpStatusCodeException e) {
-            throw new StudyException("getCaseList HttpStatusCodeException", e);
-        }
+            return studyRepository.deleteByName(studyName);
+        });
     }
 
-    UUID importCase(MultipartFile multipartFile) throws IOException {
-        HttpHeaders requestHeaders = new HttpHeaders();
-        requestHeaders.setContentType(MediaType.MULTIPART_FORM_DATA);
-
+    Mono<UUID> importCase(MultipartFile multipartFile) throws IOException {
```

Comment:
An improvement could be to use the reactive APIs to get the file so that starting the call to the case server doesn't have to wait until the whole file has been received on the study server